cmake/modules/Buildpmem.cmake

```diff
@@ -16,7 +16,7 @@ function(build_pmem)
 
   ExternalProject_Add(pmdk_ext
       GIT_REPOSITORY "https://github.com/ceph/pmdk.git"
-      GIT_TAG "1.7"
+      GIT_TAG "1.10"
```

Comment:
so we cannot use the libpmem-dev shipped by ubuntu focal anymore..

Reply:
i created #40578 to enable find_package() to specify the version of libpmem.

Reply:
> rsion of libpmem.

As this PR merged, I cherry-pick Yingxin's PR: https://github.com/ceph/ceph/pull/39154. And modified find_package() version, please review again, Thanks.

Reply:
> i created #40578 to enable find_package() to specify the version of libpmem.

As this PR merged, I cherry-pick Yingxin's PR: https://github.com/ceph/ceph/pull/39154. And modified find_package() version, please review again, Thanks.

Reply:
i'd suggest update `ceph/ceph.spec.in` to set `_system_pmdk` to 0 accordingly, as, at the time of writing, the only distro which packages libpmem 1.10 is fedora rawhide.

Reply:
also, to require pmem libraries v1.10, we will have to either

- include pmdk as a submodule, see #40186
- or include pmdk in `make-dist`.

Reply:
i'd suggest go with the 2nd approach. and use the precompiled packages while running "make check" and packaging for teuthology, see #40617

Reply:
OK, I'll try to do it in `make-dist`. But it may take some time because I'm not familiar with it

Reply:
@CongMinYin @tchaikov  I have finished the 2nd approach, please see #40866